### ISSUE TITLE
paren/curly brace identified within regular expression string

### ISSUE THREAD

**Issue by aronatkins** [jimhester/lintr#793]
```r
# file: foo/foo.R
something <- function(url) {
    gsub("([^/]*/){3}", "", url)
}
```

`lintr::lint_dir` marks this as a problem even though the characters are within a regular expression string. It is not R code.

```
> lintr::lint_dir("foo")
.
foo.R:2:18: style: There should be a space between right parenthesis and an opening curly brace.
    gsub("([^/]*/){3}", "", url)
```

**Comment by MichaelChirico** [comment]
Can you confirm you are using the most recent version of lintr?

**Comment by aronatkins** [comment]
Yes. It is the latest CRAN version.

```r
packageVersion("lintr")
# => [1] ‘2.0.1’
```

I just tested with the GitHub version using `remotes::install_github("jimhester/lintr")` and do not see this complaint still happening. Apologies for not testing this first. Didn't realize that the CRAN release was so far away from GH.
